### PR TITLE
Added scopeValue to RecordList

### PR DESCRIPTION
**Pull request by danhrrn**
https://github.com/rainlab/builder-plugin/issues/213

Added `scopeValue` to the record list component.

### CODE REVIEW

File: lang/en/lang.php [hunk 1]
```diff
@@ -653,6 +653,8 @@
         'list_scope' => 'Scope',
         'list_scope_description' => 'Optional model scope to fetch the records',
         'list_scope_default' => '--select a scope, optional--',
+        'list_scope_value' => 'Scope value',
```
Comment (danhrrn) [inline lang/en/lang.php]: Added en "Scope value" translation string

File: lang/en/lang.php [hunk 2]
```diff
@@ -653,6 +653,8 @@
...
+        'list_scope_value_description' => 'Optional model scope value to fetch the records with',
```
Comment (danhrrn) [inline lang/en/lang.php]: Added en "Scope value description" translation string

File: components/RecordList.php [hunk 3]
```diff
@@ -84,6 +84,12 @@ public function defineProperties()
                 'depends'     => ['modelClass'],
                 'showExternalParam' => false
             ],
+            'scopeValue' => [
```
Comment (danhrrn) [inline components/RecordList.php]: Added component property

File: components/RecordList.php [hunk 4]
```diff
@@ -266,9 +272,14 @@ protected function listRecords()
 
         $model = new $modelClassName();
         $scope = $this->getScopeName($model);
+        $scopeValue = $this->property('scopeValue');
```
Comment (danhrrn) [inline components/RecordList.php]: Store `scopeValue` property value for later ue

File: components/RecordList.php [hunk 5]
```diff
@@ -266,9 +272,14 @@ protected function listRecords()
...
 
         if ($scope !== null) {
-            $model = $model->$scope();
+            if ($scopeValue !== null) {
```
Comment (danhrrn) [inline components/RecordList.php]: If `scopeValue` is given, use it.

Comment (LukeTowers) [inline components/RecordList.php]: This is good thinking on your part, but unnecessary. PHP can send more arguments than the method asks for with no problems, but if the method asks for more arguments than are sent, that will cause a problem. You should be fine to just always do `$model->$scope($scopeValue)`

File: updates/version.yaml [hunk 6]
```diff
@@ -19,3 +19,4 @@
 1.0.19: Added a feature allowing to add all database columns to a list definition. Added max length validation for database table and column names.
 1.0.20: Fixes a bug where form the builder could trigger the "current.hasAttribute is not a function" error.
 1.0.21: Back-end navigation sort order updated.
+1.0.22: Add scopeValue to the Record list component.
```
Comment (LukeTowers) [inline updates/version.yaml]: Change to `Added scopeValue property to the RecordList component.`

File: lang/en/lang.php (hunk 2 above)
Comment (LukeTowers) [inline lang/en/lang.php]: change to `Optional value to pass to the model scope`

**Pull request merged**